extra/cloudformation/default-unauth-resources-template.yaml

```diff
@@ -878,55 +878,67 @@ Resources:
+                                      'GeofenceProperties': {
+                                          'stop_name': stopName
+                                      },
+                                      'Geometry': {
+                                          'Circle': {
+                                              'Center': coordinates,
+                                              'Radius': 200
+                                          }
+                                      }
                                   },
-                                'Geometry': {
-                                    'Circle': {
-                                        'Center': coordinates,
-                                        'Radius': 200
-                                    }
-                                    }
-                                },
-                            ]
-                        )
-                responseValue = 120
-                responseData = {}
-                responseData['Data'] = responseValue
-                cfnresponse.send(event, context, cfnresponse.SUCCESS, 
-                  responseData, 'AmazonLocationDemoGeofenceCreator')                        
-              if event['RequestType'] == "Delete":
-                for route in routes:
-                    id = route['id']
-                    collectionName = ('location.aws.com.demo.geofences.' + route['geofenceCollection'])
-                    geofence_collection = client.delete_geofence_collection(
-                        CollectionName=collectionName
-                    )
-                responseValue = 120
-                responseData = {}
-                responseData['Data'] = responseValue
-                cfnresponse.send(event, context, cfnresponse.SUCCESS, 
-                  responseData, 'AmazonLocationDemoGeofenceCreator')
+                              ]
+                          )
+                  responseValue = 120
+                  responseData = {'Data': responseValue}
+                  cfnresponse.send(event, context, cfnresponse.SUCCESS, responseData, 'AmazonLocationDemoGeofenceCreator')
+              
+              elif event['RequestType'] == "Delete":
+                  for route in routes:
+                      id = route['id']
+                      collectionName = ('location.aws.com.demo.geofences.' + route['geofenceCollection'])
+                      try:
+                          geofence_collection = client.delete_geofence_collection(
+                              CollectionName=collectionName
+                          )
+                      except client.exceptions.ResourceNotFoundException:
+                          pass  # If the collection does not exist, continue
+                  responseValue = 120
+                  responseData = {'Data': responseValue}
+                  cfnresponse.send(event, context, cfnresponse.SUCCESS, responseData, 'AmazonLocationDemoGeofenceCreator')
```

Comment:
Suggested on other CR as well, but this could be moved above the Create/Update if as `if event['RequestType'] in ["Delete", "Update"]:` to avoid duplicating the delete logic. The cfnresponse.send() would need to be pulled out of the if blocks though to prevent the Update case from reporting success too early.

Reply:
resolved